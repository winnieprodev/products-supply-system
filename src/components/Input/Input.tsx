import React from "react";

import * as Styled from "./input.styled";

interface OwnProps {
  value: string;
  onChange: (newValue: string) => void;
  onlyDigits?: boolean;
  maxLength?: number;
  placeholder?: string;
  className?: string;
}

const Input: React.FC<OwnProps> = ({
  value,
  onlyDigits = false,
  maxLength = 30,
  onChange,
  ...rest
}) => (
  <Styled.Input
    value={value}
    onChange={(e) => onChange(e.target.value)}
    $onlyDigits={onlyDigits}
    maxLength={maxLength}
    {...rest}
  />
);

export default Input;
