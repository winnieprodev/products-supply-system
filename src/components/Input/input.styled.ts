import styled, { css } from "styled-components";
import {
  secondaryColor,
  secondaryColorDarker,
  secondaryColorLighter,
  textOnSecondaryColor,
} from "styles/colors";

interface InputProps {
  $onlyDigits?: boolean;
  maxLength: number;
}

export const Input = styled.input<InputProps>`
  border: 0;
  font-size: 12px;
  background-color: ${secondaryColor};
  color: ${textOnSecondaryColor};
  border-radius: 6px;
  padding: 8px 12px;
  outline: none;
  box-shadow: inset 1px 2px 8px rgb(0, 0, 0, 0.05);
  transition: background ease 0.3s;
  width: ${({ maxLength }) => maxLength * 10}px;
  ${({ $onlyDigits }) =>
    $onlyDigits &&
    css`
      text-align: right;
    `}

  &::placeholder {
    color: #bdbdbd;
  }

  &:hover {
    box-shadow: inset 2px 3px 8px rgb(0, 0, 0, 0.1);
    background-color: ${secondaryColorLighter};
  }
  &:active {
    background-color: ${secondaryColorDarker};
  }
`;
