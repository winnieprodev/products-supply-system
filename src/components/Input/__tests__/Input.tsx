import { render } from "@testing-library/react";

import Input from "../Input";

describe("Input", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<Input value="1" onChange={jest.fn()} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for only digits and max length", () => {
      const { container } = render(
        <Input value="1" onChange={jest.fn()} maxLength={1} onlyDigits />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
