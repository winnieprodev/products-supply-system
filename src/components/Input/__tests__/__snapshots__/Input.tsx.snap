// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot tests it matches the snapshot 1`] = `
.c0 {
  border: 0;
  font-size: 12px;
  background-color: #EEEEEE;
  color: #5F5F5F;
  border-radius: 6px;
  padding: 8px 12px;
  outline: none;
  box-shadow: inset 1px 2px 8px rgb(0,0,0,0.05);
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  width: 300px;
}

.c0::-webkit-input-placeholder {
  color: #bdbdbd;
}

.c0::-moz-placeholder {
  color: #bdbdbd;
}

.c0:-ms-input-placeholder {
  color: #bdbdbd;
}

.c0::placeholder {
  color: #bdbdbd;
}

.c0:hover {
  box-shadow: inset 2px 3px 8px rgb(0,0,0,0.1);
  background-color: #F3F3F3;
}

.c0:active {
  background-color: #E7E7E7;
}

<input
  class="c0"
  maxlength="30"
  value="1"
/>
`;

exports[`Input Snapshot tests it matches the snapshot for only digits and max length 1`] = `
.c0 {
  border: 0;
  font-size: 12px;
  background-color: #EEEEEE;
  color: #5F5F5F;
  border-radius: 6px;
  padding: 8px 12px;
  outline: none;
  box-shadow: inset 1px 2px 8px rgb(0,0,0,0.05);
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  width: 10px;
  text-align: right;
}

.c0::-webkit-input-placeholder {
  color: #bdbdbd;
}

.c0::-moz-placeholder {
  color: #bdbdbd;
}

.c0:-ms-input-placeholder {
  color: #bdbdbd;
}

.c0::placeholder {
  color: #bdbdbd;
}

.c0:hover {
  box-shadow: inset 2px 3px 8px rgb(0,0,0,0.1);
  background-color: #F3F3F3;
}

.c0:active {
  background-color: #E7E7E7;
}

<input
  class="c0"
  maxlength="1"
  value="1"
/>
`;
